### PR TITLE
Enable bucket ownership for content and log bucket

### PR DESCRIPTION
**Pull request by Jlrine2**
Blog post for this feature: https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-s3-object-ownership-simplify-access-management-data-s3/

We previously granted permissions for S3 Access Logs to be written to the log bucket via ACLs. This change requires we grant those permissions via a policy instead. See "Grant permissions to the logging service principal using a bucket policy" at https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html

### CODE REVIEW

File: apps/main-cf.yml [hunk 1]
```diff
@@ -214,6 +214,28 @@ Resources:
...
+        Statement:
+            - Effect: Allow
+              Principal:
+                service: logging.s3.amazonaws.com
+              Action:
+                - s3:PutObject
+              Resource: !GetAtt LogBucket.Arn
+              Condition:
+                ArnLike:
+                  "aws:SourceArn": !Sub "arn:aws:s3:::${ContentBucket.Arn}"
```
Comment (asjohnston-asf) [inline apps/main-cf.yml:236]:
```suggestion
                  "aws:SourceArn": !GetAtt ContentBucket.Arn
```

File: apps/main-cf.yml [hunk 2]
```diff
@@ -214,6 +214,28 @@ Resources:
...
+      Bucket: !Ref LogBucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+            - Effect: Allow
+              Principal:
+                service: logging.s3.amazonaws.com
+              Action:
+                - s3:PutObject
+              Resource: !GetAtt LogBucket.Arn
```
Comment (asjohnston-asf) [inline apps/main-cf.yml:233]:
```suggestion
              Resource: !Sub "${LogBucket.Arn}/*"
```

File: apps/main-cf.yml [hunk 3]
```diff
@@ -214,6 +214,28 @@ Resources:
...
+    Properties:
+      Bucket: !Ref LogBucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+            - Effect: Allow
+              Principal:
+                service: logging.s3.amazonaws.com
+              Action:
+                - s3:PutObject
```
Comment (asjohnston-asf) [inline apps/main-cf.yml:232]:
```suggestion
              Action: s3:PutObject
```

**Comment by jhkennedy** [comment]
waiting on a cfn-lint release to fix failing test

**Comment by asjohnston-asf** [comment]
Still waiting on a resolution to https://github.com/aws-cloudformation/cfn-lint/issues/2191 and a subsequent `cfn-lint` release

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**